cvat/apps/engine/annotation.py

```diff
@@ -1222,241 +662,43 @@ def init_from_db(self):
-                            dump_data["z_order"] = str(shape["z_order"])
-
-                        if shape["type"] == models.ShapeType.RECTANGLE:
-                            dumper.open_box(dump_data)
-                        elif shape["type"] == models.ShapeType.POLYGON:
-                            dumper.open_polygon(dump_data)
-                        elif shape["type"] == models.ShapeType.POLYLINE:
-                            dumper.open_polyline(dump_data)
-                        elif shape["type"] == models.ShapeType.POINTS:
-                            dumper.open_points(dump_data)
-                        else:
-                            raise NotImplementedError("unknown shape type")
-
-                        for attr in shape.get("attributes", []) + track.get("attributes", []):
-                            db_attribute = db_attribute_by_id[attr["spec_id"]]
-                            dumper.add_attribute(OrderedDict([
-                                ("name", db_attribute.name),
-                                ("value", attr["value"])
-                            ]))
-
-                        if shape["type"] == models.ShapeType.RECTANGLE:
-                            dumper.close_box()
-                        elif shape["type"] == models.ShapeType.POLYGON:
-                            dumper.close_polygon()
-                        elif shape["type"] == models.ShapeType.POLYLINE:
-                            dumper.close_polyline()
-                        elif shape["type"] == models.ShapeType.POINTS:
-                            dumper.close_points()
-                        else:
-                            raise NotImplementedError("unknown shape type")
-                    dumper.close_track()
-            dumper.close_root()
+            self._merge_data(annotation.ir_data, start_frame, overlap)
+
+    def dump(self, file_path, scheme, host, dumper, spec):
+        anno_exporter = Annotation(
+            annotation_ir=self.ir_data,
+            db_task=self.db_task,
+            scheme=scheme,
+            host=host,
+            )
+
+        with open(file_path, 'wb') as dump_file:
+            source_code = open(os.path.join(settings.BASE_DIR, dumper.handler_file.name)).read()
+            global_vars = globals()
+            imports = import_modules(source_code)
+            global_vars.update(imports)
+            exec(source_code, global_vars)
+            global_vars["annotations"] = anno_exporter
+            global_vars["dump_spec"] = spec
+            global_vars["file_object"] = dump_file
+
+            exec("dump(file_object, annotations, dump_spec)", global_vars)
+
+    def upload(self, file_object, parser, spec):
+        anno_importer = Annotation(
+            annotation_ir=AnnotationIR(),
+            db_task=self.db_task,
+            create_callback=self.create,
+            )
+        self.delete()
+        _parse_task_annotation(file_object, anno_importer, parser, spec)
```

Comment:
Please remove _parse_task_annotation function and add its source code here as you did for `dump`.